metric_providers/powermetrics/provider.py

```diff
@@ -35,43 +41,44 @@ def __init__(self, resolution, skip_check=False):
             '-o',
             self._filename]
 
+
     def check_system(self):
-            if self.is_powermetrics_running():
-                raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\nPlease close it before running the Green Metrics Tool.')
+        if self._pm_process_count > 0:
+            raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\n'
+                                                   'Please close it before running the Green Metrics Tool.\n'
+                                                   'You can also override this with --skip-system-checks\n')
 
-    def is_powermetrics_running(self):
-        ps = subprocess.run(['pgrep', '-qx', 'powermetrics'], check=False)
-        if ps.returncode == 1:
-            return False
-        return True
+    def is_our_powermetrics_running(self):
+        pw_processes = subprocess.check_output(['pgrep', '-ix', 'powermetrics'], text=True)
+        total_count = len(pw_processes.strip().split('\n')) if pw_processes else 0
+        minus_startup = total_count -  self._pm_process_count
+        return  minus_startup >= 1
 
 
     def stop_profiling(self):
         try:
             # We try calling the parent method but if this doesn't work we use the more hardcore approach
             super().stop_profiling()
         except PermissionError:
-            #This isn't the nicest way of doing this but there isn't really any other way that is nicer
+            # This isn't the nicest way of doing this but there isn't really any other way that is nicer that doesn't
+            # open up a huge security hole
             subprocess.check_output('sudo /usr/bin/killall powermetrics', shell=True)
             print('Killed powermetrics process with killall!')
+            if self._pm_process_count > 0:
+                print('----------------------------------------------------------------------------------------------')
+                print('This means we will have also killed the already running powermetrics process. Please restart!')
```

Comment:
Suggestion as I think it is confusing what exactly to restart and also referencing "the process" ... that implies too much knowlege: 
"This means we will have also killed any other already running powermetrics process. Please restart them if needed!